src/storage/oid.h

```diff
@@ -133,24 +136,28 @@
    ((((unsigned int)(oidp)->pageid) >> 8) | \
     (((unsigned int)(oidp)->volid) << 24)))
 
-/* wrapper functinos to be used as a callable object in c++ */
-inline bool
-oid_lt (const OID * oidp1, const OID * oidp2)
+#ifdef __cplusplus
+// *INDENT-OFF*
+template <>
+struct std::hash<OID>
 {
-  return OID_LT (oidp1, oidp2);
-}
+  size_t operator()(const OID& oid) const
+  {
+    return OID_PSEUDO_KEY (&oid);
+  }
+};
 
-inline bool
-oid_eq (const OID * oidp1, const OID * oidp2)
+inline bool operator==(const OID& oid1, const OID& oid2)
 {
-  return OID_EQ (oidp1, oidp2);
+  return OID_EQ (&oid1, &oid2);
 }
 
-inline bool
-oid_pseudo_key (const OID * oidp)
+inline bool operator!=(const OID& oid1, const OID& oid2)
 {
-  return OID_PSEUDO_KEY (oidp);
+  return !(oid1 == oid2);
```

Comment:
How about '!OID_EQ (&oid1, &oid2)'?

Reply:
Could you let me know why you think it's better?

Reply:
I just think it is more readable. And It is consistent because the !OID_EQ() is used in many parts. Is there any benefit to your style? Are there any standards or rules to check by me?

Reply:
The rule is just that "When defining ==, also define != and make sure that a!=b means !(a==b)".
 I thought it would be easy to keep this rule if I define `==` and write `!=` like `!(oid1 == oid2)` so that if needed to change, I can just modify `==`.

Reply:
It is not an important issue. So it depends on you.

Reply:
I think your suggestion also makes sense. I'll follow it.